superset/views/core.py

```diff
@@ -2496,10 +2502,10 @@ def sql_json(self):
             )
 
         client_id = request.form.get('client_id') or utils.shortid()[:10]
-
+        limits = [mydb.db_engine_spec.get_limit_from_sql(sql), limit]
         query = Query(
             database_id=int(database_id),
-            limit=mydb.db_engine_spec.get_limit_from_sql(sql),
+            limit=min(lim for lim in limits if lim is not None),
```

Comment:
I built this with the assumption that we are going to do limiting via query replacement within the context of SQL Lab, so only the smaller limit is chosen.